src/main/scala/com/github/conduktor/ksm/source/SourceAclResult.scala

```diff
@@ -1,15 +1,30 @@
 package com.github.conduktor.ksm.source
 
-import com.github.conduktor.ksm.parser.CsvParserException
+import cats.kernel.Monoid
+import com.github.conduktor.ksm.parser.ParserException
 import com.github.conduktor.ksm.source.SourceAclResult.{
-  ksmAcls,
-  ParsingExceptions
+  ParsingExceptions,
+  ksmAcls
 }
 import kafka.security.auth.{Acl, Resource}
 
 object SourceAclResult {
```

Comment:
In Scala, we usually build an ADT (you can find a lot of documentation on the web) to express different status.
Here, I would write:
```
sealed trait SourceAclResult
case class Success(result) extends SourceAclResult
case class Failure(errors: List[ParserException]) extends SourceAclResult
```
you can keep the monoid, of course, but it will be much easier to read

Reply:
I used this pattern to make it possible to easily use combineAll to merge the list of ACLs, can that be done with that approach?

If yes could you suggest how?

Reply:
maybe something like that would work:
```
implicit val monoid: Monoid[SourceAclResult] = new Monoid[SourceAclResult] {
    override def empty: Success = Success.empty
    override def combine(
        x: SourceAclResult,
        y: SourceAclResult
    ): SourceAclResult = (x, y) match {
      case (Success(_), Failure(errors)) =>
        Failure(errors)
      case (Failure(errors), Success(_)) =>
        Failure(errors)
      case (Failure(errors1), Failure(errors2)) =>
        Failure(errors1 ++ errors2)
      case (Success(result1), Success(result2)) =>
        Success(result1 ++ result2)
  }
```

Reply:
I tried various options with ADTs but that approach was making the whole thing more complicated in some cases because I don't want to pattern match always. For example in tests I would had to use asInstanceOf I guess. I also tried a leaner approach by defining SourceAclResult as a Either[ParsingExceptions, KsmAcls] but that broke my monoid extension and I did not know how to fix it (error where not combined anymore).

In the end I simply refactored the monoid implementation to be a little more readable and I think it looks pleasant now.